packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -366,7 +385,7 @@ export function getInternalReactConstants(
         if (elementType.displayName) {
           return elementType.displayName;
         } else {
-          return getDisplayName(type, 'Anonymous');
+          return getDisplayName(resolvedType, 'Anonymous');
```

Comment:
I had been thinking of suggesting a different approach that just kind of localized the check:
```diff
diff --git a/packages/react-devtools-shared/src/backend/renderer.js b/packages/react-devtools-shared/src/backend/renderer.js
index 761c78935..284183f9f 100644
--- a/packages/react-devtools-shared/src/backend/renderer.js
+++ b/packages/react-devtools-shared/src/backend/renderer.js
@@ -365,6 +365,9 @@ export function getInternalReactConstants(
       case SimpleMemoComponent:
         if (elementType.displayName) {
           return elementType.displayName;
+        } else if (type.type != null && type.type.render != null) {
+          // Special case of a forwardRef() nested in a memo()
+          return getDisplayName(type.type.render, 'Anonymous');
         } else {
           return getDisplayName(type, 'Anonymous');
         }
```
I don't know that I really have a preference between the two approaches I guess. 😄 Yours is fine~

Reply:
Tests added #17283